tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaProducerConstructorIntegration.cs

```diff
@@ -25,52 +27,61 @@ namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Kafka;
 [EditorBrowsable(EditorBrowsableState.Never)]
 public class KafkaProducerConstructorIntegration
 {
-    internal static CallTargetState OnMethodBegin<TTarget, TProducerBuilder>(TTarget instance, TProducerBuilder consumer)
+    internal static CallTargetState OnMethodBegin<TTarget, TProducerBuilder>(TTarget instance, TProducerBuilder? builder)
         where TProducerBuilder : IProducerBuilder
     {
-        if (Tracer.Instance.Settings.IsIntegrationEnabled(KafkaConstants.IntegrationId))
+        if (instance is null || builder is null || builder.Instance is null || builder.Config is null)
         {
-            string bootstrapServers = null;
-            var deliveryReportsEnabled = true;
-            foreach (var kvp in consumer.Config)
+            return CallTargetState.GetDefault();
+        }
+
+        if (!Tracer.Instance.Settings.IsIntegrationEnabled(KafkaConstants.IntegrationId))
+        {
+            return CallTargetState.GetDefault();
+        }
```

Comment:
WDYT about merging these two if's? Since if the integration not enabled then no point on doing anything else etc..

Reply:
Yeah makes sense to me
I'll swap their order so the integration check gets hit first as well